homeassistant/components/ihc/__init__.py

```diff
@@ -0,0 +1,122 @@
+"""IHC platform."""
+# pylint: disable=too-many-arguments, too-many-instance-attributes
+import time
+import logging
+import os.path
+import xml.etree.ElementTree
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_URL, CONF_USERNAME, CONF_PASSWORD
+from homeassistant.config import load_yaml_config_file
+
+from homeassistant.components.ihc import const
+
+REQUIREMENTS = ['ihcsdk==2.1.0']
+DOMAIN = 'ihc'
+
+CONF_INFO = 'info'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_URL): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_INFO): cv.boolean
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """Setyp the IHC platform."""
+    from ihcsdk.ihccontroller import IHCController
+    url = config[DOMAIN].get(CONF_URL)
```

Comment:
Please cache the first lookup in a variable to be able to reuse it multiple times below.
```py
conf = config[DOMAIN]
url = conf.get(CONF_URL)
```

Reply:
Done